x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/RBACEngine.java

```diff
@@ -141,6 +139,11 @@ public void resolveAuthorizationInfo(RequestInfo requestInfo, ActionListener<Aut
         );
     }
 
+    @Override
+    public void resolveAuthorizationInfo(Subject subject, ActionListener<AuthorizationInfo> listener) {
+        rolesStore.getRole(subject, listener.map(role -> new RBACAuthorizationInfo(role, role)));
+    }
```

Comment:
Not related to this PR, but this usage shows why I didn't remove the `AnonymousUser` argument from `Subject.getRoleReferenceIntersection`. As in its current state, this works correctly because anonymous user roles are handled inside `rolesStore.getRole` and the caller does not have to think about it. If we remove it, this means the subject must be prepared with anonymous roles before calling to this method which could be trappy.